components/encryption/src/manager/mod.rs

```diff
@@ -318,6 +379,35 @@ impl Dicts {
     }
 }
 
+fn current_data_key(key_dict: &KeyDictionary) -> (u64, &DataKey) {
+    (
+        key_dict.current_key_id,
+        key_dict
+            .keys
+            .get(&key_dict.current_key_id)
+            .unwrap_or_else(|| {
+                panic!(
+                    "current data key not found! Number of keys {}",
+                    key_dict.keys.len()
+                );
+            }),
+    )
+}
+
+fn get_index(fname: &str) -> usize {
+    let mut hasher = DefaultHasher::new();
```

Comment:
1. cache the hasher with `Dicts`?
2. Shall we consider using `RandomState` to generate a new hasher? It would make the hash different from that of sub-`sharded_file_dict` maps. Maybe it could make the sub-maps hash better.